src/bidiMapper/domains/network/NetworkRequest.ts

```diff
@@ -182,13 +188,31 @@ export class NetworkRequest {
   onRequestWillBeSentEvent(event: Protocol.Network.RequestWillBeSentEvent) {
     this.#request.info = event;
     this.#queueBeforeRequestSentEvent();
+    if (
+      this.#networkStorage.requestBlockedBy(
+        this,
+        Network.InterceptPhase.BeforeRequestSent
+      ).size &&
+      this.#interceptPhase !== Network.InterceptPhase.BeforeRequestSent
+    ) {
+      return;
+    }
     this.#emitEventsIfReady();
   }
 
   onRequestWillBeSentExtraInfoEvent(
     event: Protocol.Network.RequestWillBeSentExtraInfoEvent
   ) {
     this.#request.extraInfo = event;
+    if (
+      this.#networkStorage.requestBlockedBy(
```

Comment:
I think we should use the CdpTarget to find out if interception is enabled, it's the source of truth if an even is going to be sent.

Reply:
It's not about the interception being enabled, it's about knowing if the request should be intercepted irrespective of the underlying CDP status